src/Workspaces/Core/Portable/Storage/SQLite/v2/SQLitePersistentStorage_Threading.cs

```diff
@@ -28,12 +29,30 @@ internal partial class SQLitePersistentStorage
         // Read tasks go to the concurrent-scheduler where they can run concurrently with other read
         // tasks.
         private Task<TResult> PerformReadAsync<TArg, TResult>(Func<TArg, TResult> func, TArg arg, CancellationToken cancellationToken) where TArg : struct
-            => PerformTaskAsync(func, arg, _connectionPoolService.Scheduler.ConcurrentScheduler, cancellationToken);
+        {
+            // Suppress ExecutionContext flow for asynchronous operations that write to the database. In addition to
+            // avoiding ExecutionContext allocations, this clears the LogicalCallContext and avoids the need to clone
+            // data set by CallContext.LogicalSetData at each yielding await in the task tree.
+            //
+            // ⚠ DO NOT AWAIT INSIDE THE USING. The Dispose method that restores ExecutionContext flow must run on the
+            // same thread where SuppressFlow was originally run.
+            using var _ = FlowControlHelper.TrySuppressFlow();
+            return PerformTaskAsync(func, arg, _connectionPoolService.Scheduler.ConcurrentScheduler, cancellationToken);
```

Comment:
so, just for clarity. it's ok that we're going to return a task here, have the IDIsposable disposed, and hten have the caller to whatever it wants with that task?  this is just about setting some values that are only relevant for when teh task is created?

Reply:
Yes, the behavior you describe is required for correctness. It's strange and a red flag, hence the warning.